### DISCUSSION TITLE
Color to String

### DISCUSSION THREAD

**Discussion by Jaydar** [slint-ui/slint#4272] [Q&A]
How to convert between Color and String, for instance, if a TextInput inputs #EEEEEE, using this color to modify the background color of a component

**Comment by ogoffart** [comment]
This would need to be done in native code.
The most convenient for doing things in native code is to have a global.  
for example

```
export global Logic { 
    pure callback color-to-string(color) -> string;
    pure callback string-to-color(string) -> color;
}
```

And then implement these function in native code: see https://slint.dev/releases/1.3.2/docs/slint/src/recipes/recipes#global-callbacks

**Reply by Jaydar** [reply]
Thanks
